### PR TITLE
Add completion for behaviours

### PR DESCRIPTION
**Pull request by telezhnaya**
I know about tests (I have not written it yet). I just want to know, is it the same that you are waiting for.

### CODE REVIEW

**Comment by deadok22** [comment]
You've created an index of ALL modules - something that we already have in ```org.intellij.erlang.index.ErlangModuleIndex```. Moreover, you're iterating through all index entries, and filter behaviours out. What's the point of having all other entries in this index?

Also, you've added ```org.intellij.erlang.stubs.ErlangFileStub#myIsBehaviour``` which renders useless, once you create a proper index.

**Comment by telezhnaya** [comment]
I think now it is better.
but I still need org.intellij.erlang.stubs.ErlangFileStub#myIsBehaviour.

**Comment by deadok22** [comment]
Indeed, ```org.intellij.erlang.stubs.ErlangFileStub#myIsBehaviour``` is mandatory.

There are several things I don't like, though:
* You have your index keyed with behavior names, although you never query it by these names.
* Lookup elements you produce for completion are string-based, as opposed to module element-based. Having lookup elements module element-based would allow for navigation, quick-doc and stuff like that (try Ctrl+Q or Ctrl+Shift+I in completion list).
* Completion for behaviours doesn't work when there's no input: ```-behaviour(<caret>).```

**Comment by deadok22** [comment]
Oh, one more thing. You have to treat modules exporting ```behaviour_info/1``` as behaviour modules, too.

**Comment by telezhnaya** [comment]
Have a look, please.

**Comment by telezhnaya** [comment]
Check it please.

**Comment by telezhnaya** [comment]
Check it please.

File: src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package org.intellij.erlang.stubs.index;
+
+import com.intellij.psi.stubs.StringStubIndexExtension;
+import com.intellij.psi.stubs.StubIndexKey;
+import org.intellij.erlang.psi.ErlangModule;
+import org.jetbrains.annotations.NotNull;
+
+public class ErlangBehaviourModuleIndex extends StringStubIndexExtension<ErlangModule> {
+  public static final String INDEX = "behaviour";
```
Comment (deadok22) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: In previous review I advised not to expose a key which you use in module index. You did expose it anyway.

Not only that, you also use this constant for module insertion handler which breaks completion for quoted behaviour names. Please fix it and test it properly.

**Comment by telezhnaya** [comment]
I want to talk with you about it. I have fixed all, but I am not sure that this is what you want.

**Comment by telezhnaya** [comment]
Check it please.

File: src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return super.getVersion() + VERSION;
+  }
+
+  @NotNull
+  public StubIndexKey<String, ErlangModule> getKey() {
+    return KEY;
+  }
+
+  @NotNull
+  public static Collection<ErlangModule> getAllBehaviourModules(Project project) {
```
Comment (deadok22) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: Nullability annotation.

Also, I'd rather have a search scope passed here, than a project - may be, after all, suggesting modules from all over the project isn't completely correct - you cannot reference behaviours declared in IntelliJ modules which are not in current module's dependencies.

Comment (telezhnaya) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: Are you sure that we don't need project here?
I think they both are necessary.

**Comment by telezhnaya** [comment]
Check it please.

File: src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import com.intellij.psi.stubs.StubIndexKey;
+import org.intellij.erlang.psi.ErlangModule;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+
+public class ErlangBehaviourModuleIndex extends StringStubIndexExtension<ErlangModule> {
+  private static final String KEY_NAME = "erlang.behaviour.module";
+  public static final StubIndexKey<String, ErlangModule> KEY = StubIndexKey.createIndexKey(KEY_NAME);
+  public static final int VERSION = 0;
```
Comment (deadok22) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: Members ordering - move private constants below public ones.

File: src/org/intellij/erlang/psi/impl/ErlangFileImpl.java [hunk 4]
```diff
@@ -419,7 +419,8 @@ public boolean isNoAutoImportAll() {
 
   @Override
   public boolean isBehaviour() {
-    return !getCallbackMap().isEmpty() || getFunction("behaviour_info", 1) != null;
+    ErlangFunction function = getFunction("behaviour_info", 1);
+    return  !getCallbackMap().isEmpty() || function != null && function.isExported();
```
Comment (deadok22) [inline src/org/intellij/erlang/psi/impl/ErlangFileImpl.java]: This method should first attempt to get it's result from a file stub.

File: src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java (hunk 3 above)
Comment (telezhnaya) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: I use KEY_NAME at next line after declaration. 
That would be illegal forward reference.

Comment (deadok22) [inline src/org/intellij/erlang/stubs/index/ErlangBehaviourModuleIndex.java]: Oh, I see. You're right.

**Comment by deadok22** [comment]
Completion for behaviours with names starting with a capital letter doesn't work:
```erlang
-behaviour(B<caret>).
```

**Pull request merged**